airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/operation/Operation.kt

```diff
@@ -11,7 +11,16 @@ import io.airbyte.protocol.models.v0.AirbyteMessage
  * OperationType}s and proxies to an {@link OperationExecutor} that performs the actual work.
  */
 interface Operation {
+
     fun type(): OperationType
 
     fun execute(): Result<AirbyteMessage?>
+
+    fun requiresCatalog(): Boolean {
+        return false
+    }
+
+    fun requiresConfiguration(): Boolean {
+        return false
+    }
```

Comment:
These methods might better be replaced by attributes on the `OperationType` enum values.

Reply:
Let me be stronger by replacing "might" with "would" after looking at the `requiresConfiguration` implementation in `CatalogValidator`.

Reply:
I have moved these flags to the enum.